dlang/source/flatbuffers/flatbufferbuilder.d

```diff
@@ -0,0 +1,301 @@
+		{
+			auto oldBufSize = cast(int)_buffer.length;
+			growBuffer();
+			_space += cast(int)_buffer.length - oldBufSize;
+		}
+		if(alignSize > 0)
+			pad(alignSize);
+	}
+
+	void put(T)(T x) if(is(T == bool) || is(T == byte) || is(T == ubyte) ||
+		is(T == short) || is(T == ushort) || is(T == int) || is(T == uint) || is(T == long) 
+		|| is(T == ulong) || is(T == float) || is(T == double)) 
+	{
+		static if (is(T == bool)) {
+			_space -= 1;
+		} else {
+			_space -= T.sizeof;
+		}
+		_buffer.put!T(_space, x);
+	}
+
+	///Adds a scalar to the buffer, properly aligned, and the buffer grown if needed.
+	void addBool(bool x) { prep(byte.sizeof, 0); put!bool(x); }
+	void addByte(byte x) { prep(byte.sizeof, 0); put!byte(x); }
+	void addUbyte(ubyte x) { prep(ubyte.sizeof, 0); put!ubyte(x); }
+	void addShort(short x) { prep(short.sizeof, 0); put!short(x); }
+	void addUshort(ushort x) { prep(ushort.sizeof, 0); put!ushort(x); }
+	void addInt(int x) { prep(int.sizeof, 0); put!int(x); }
+	void addUint(uint x) { prep(uint.sizeof, 0); put!uint(x); }
+	void addLong(long x) { prep(long.sizeof, 0); put!long(x); }
+	void addUlong(ulong x) { prep(ulong.sizeof, 0); put!ulong(x); }
+	void addFloat(float x) { prep(float.sizeof, 0); put!float(x); }
+	void addDouble(double x) { prep(double.sizeof, 0); put!double(x); }
+	
+	///Adds on offset, relative to where it will be written.
+	void addOffset(int off)
+	{
+		prep(int.sizeof, 0); //Ensure alignment is already done.
+		if(off > offset())
+			throw new ArgumentException("FlatBuffers: must be less than offset.", "off");
+		
+		off = offset() - off + cast(int)int.sizeof;
+		put!int(off);
+	}
+
+	void startVector(int elemSize, int count, int alignment)
+	{
+		notNested();
+		_vectorNumElems = count;
+		prep(int.sizeof, elemSize * count);
+		prep(alignment, elemSize * count); //Just in case alignment > int.
+	}
+	
+	int endVector()
+	{
+		put!int(_vectorNumElems);
+		return offset();
+	}
+	
+	void nested(int obj)
+	{
+		//Structs are always stored inline, so need to be created right
+		//where they are used. You'll get this assert if you created it
+		//elsewhere.
+		if(obj != offset())
+			throw new Exception("FlatBuffers: struct must be serialized inline.");
+	}
+	
+	void notNested()
+	{
+		//You should not be creating any other objects or strings/vectors
+		//while an object is being constructed.
+		if(_vtable)
+			throw new Exception("FlatBuffers: object serialization must not be nested.");
+	}
+	
+	void startObject(int numfields)
+	{
+		notNested();
+		_vtable = new int[](numfields);
+		_objectStart = offset();
+	}
+	
+	///Set the current vtable at `voffset` to the current location in the buffer.
+	void slot(int voffset)
+	{
+		_vtable[voffset] = offset();
+	}
+	
+	///Add a scalar to a table at `o` into its vtable, with value `x` and default `d`.
+	void addBool(int o, bool x, bool d) { if(x != d) { addBool(x); slot(o); } }
+	void addByte(int o, byte x, byte d) { if(x != d) { addByte(x); slot(o); } }
```

Comment:
you need some `/// ditto` here and on some other places to apply the documentation to all the members